fiftyone/factory/repos/delegated_operation.py

```diff
@@ -310,9 +313,15 @@ def update_run_state(
             update["$set"]["status"] = progress
             update["$set"]["status"]["updated_at"] = datetime.utcnow()
 
+        # Using pipeline update instead of a single update doc fixes a case
```

Comment:
Since the update can either be a pipeline or a single update, why do we need to only make it a pipeline under certain conditions and not just always like the current code? Is there a different bug that can happen by consistently using pipeline for all cases?

Reply:
It's nonstandard. I'd prefer to use the standard approach and deviate only when necessary.